core/scripts/install.sh

```diff
@@ -122,16 +122,32 @@ if (empty($options['db-url'])) {
   exit;
 }
 $url = parse_url($options['db-url']);
-$url += array(
-  'driver' => NULL,
-  'user' => NULL,
-  'pass' => NULL,
-  'host' => NULL,
-  'port' => NULL,
-  'path' => NULL,
-  'database' => NULL,
-);
+
 $url = (object)array_map('urldecode', $url);
+
+// Assign null values to all required fields that do not already exist
+if(!isset($url->driver) && empty($url->driver)) {
+	$url->driver = NULL ;
+}
+if(!isset($url->user) && empty($url->user)) {
+	$url->user = NULL ;
+}
+if(!isset($url->pass) && empty($url->pass)) {
+	$url->pass = NULL ;
+}
+if(!isset($url->host) && empty($url->host)) {
+	$url->host = NULL ;
+}
+if(!isset($url->port) && empty($url->port)) {
+	$url->port = NULL ;
+}
+if(!isset($url->path) && empty($url->path)) {
+	$url->path = NULL ;
+}
+if(!isset($url->database) && empty($url->database)) {
+	$url->database = NULL ;
+}
+
```

Comment:
Instead of all the above, I suggest a more compact version (starting after `$url = parse_url($options['db-url']);`)

```php
$url = array_map('urldecode', $url);
$url += array(
  'driver' => NULL,
  'user' => NULL,
  'pass' => NULL,
  'host' => NULL,
  'port' => NULL,
  'path' => NULL,
  'database' => NULL,
);

$url = (object)$url;
```

This avoids all the `if` checks, and avoids sending NULLs to `urlencode`

Reply:
The next lines do not use `$url->database`. Does it make sense to set it to `NULL`?